controllers/machine_controller.go

```diff
@@ -339,32 +350,25 @@ func (r *MachineReconciler) drainNode(cluster *clusterv1.Cluster, nodeName strin
 
 func (r *MachineReconciler) deleteNode(ctx context.Context, cluster *clusterv1.Cluster, name string) error {
 	logger := r.Log.WithValues("machine", name)
-
-	if cluster == nil {
-		// Try to retrieve the Node from the local cluster, if no Cluster reference is found.
-		var node corev1.Node
-		if err := r.Client.Get(ctx, client.ObjectKey{Name: name}, &node); err != nil {
-			return err
+	// Retrieve the Node from the local cluster
+	c := r.Client
+	if cluster != nil {
+		// If a Cluster reference is found, proceed to get the remote Cluster client to get the Node
+		// TODO refactor as part of https://github.com/kubernetes-sigs/cluster-api/issues/1667
+		var err error
+		logger = logger.WithValues("cluster", cluster.Name, "namespace", cluster.Namespace)
+		c, err = remote.NewClusterClient(r.Client, cluster, r.scheme)
+		if err != nil {
+			logger.Error(err, "Error creating a remote client for cluster while deleting Machine, won't retry")
+			return nil
 		}
-		return r.Client.Delete(ctx, &node)
 	}
 
-	logger = logger.WithValues("cluster", cluster.Name, "namespace", cluster.Namespace)
-
-	// Otherwise, proceed to get the remote cluster client and get the Node.
-	remoteClient, err := remote.NewClusterClient(r.Client, cluster)
-	if err != nil {
-		logger.Error(err, "Error creating a remote client for cluster while deleting Machine, won't retry")
-		return nil
+	node := &corev1.Node{}
+	if err := c.Get(ctx, client.ObjectKey{Name: name}, node); err != nil {
```

Comment:
We don't need to do a `Get` first. If you fill in the node like this, the delete works (I tested this with a ConfigMap):
```go
node := &corev1.Node{
  ObjectMeta: metav1.ObjectMeta{
    Name: name,
  },
}
```